delta/sdk-testkit/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/testkit/ResourcesDummy.scala

```diff
@@ -0,0 +1,178 @@
+  )(implicit caller: Subject): IO[ResourceRejection, DataResource] =
+    eval(TagResource(id, project, schemaOpt, tagRev, tag, rev, caller))
+
+  override def deprecate(
+      id: Iri,
+      project: ProjectRef,
+      schemaOpt: Option[ResourceRef],
+      rev: Long
+  )(implicit caller: Subject): IO[ResourceRejection, DataResource] =
+    eval(DeprecateResource(id, project, schemaOpt, rev, caller))
+
+  override def fetch(id: Iri, project: ProjectRef, schemaOpt: Option[ResourceRef]): UIO[Option[DataResource]] =
+    cache.fetch((project, id)).map(validateSameSchema(_, schemaOpt))
+
+  override def fetchAt(
+      id: Iri,
+      project: ProjectRef,
+      schemaOpt: Option[ResourceRef],
+      rev: Long
+  ): IO[RevisionNotFound, Option[DataResource]] =
+    journal
+      .stateAt((project, id), rev, Initial, Resources.next, RevisionNotFound.apply)
+      .map(_.flatMap(_.toResource))
+      .map(validateSameSchema(_, schemaOpt))
+
+  private def eval(cmd: ResourceCommand): IO[ResourceRejection, DataResource] =
+    semaphore.withPermit {
+      for {
+        state <- journal.currentState((cmd.project, cmd.id), Initial, Resources.next).map(_.getOrElse(Initial))
+        event <- Resources.evaluate(fetchSchema)(state, cmd)
+        _     <- journal.add(event)
+        res   <- IO.fromEither(Resources.next(state, event).toResource.toRight(UnexpectedInitialState(cmd.id)))
+        _     <- cache.setToCache(res)
+      } yield res
+    }
+
+  override def events(offset: Offset): fs2.Stream[Task, Envelope[ResourceEvent]] =
+    journal.events(offset)
+
+  override def currentEvents(offset: Offset): fs2.Stream[Task, Envelope[ResourceEvent]] =
+    journal.currentEvents(offset)
+
+  private def sourceAsJsonLD(
+      project: Project,
+      source: Json
+  ): IO[ResourceRejection, (Iri, CompactedJsonLd, ExpandedJsonLd)] =
+    for {
+      expandedNoId <- JsonLd.expand(source).leftMap(err => InvalidJsonLdFormat(None, err))
+      id           <- expandedNoId.rootId.asIri.fold(uuidF().map(uuid => project.base / uuid.toString))(IO.pure)
+      expanded      = expandedNoId.replaceId(id)
+      compacted    <- expanded.toCompacted(source).leftMap(err => InvalidJsonLdFormat(Some(id), err))
+    } yield (id, compacted, expanded)
+
+  private def sourceAsJsonLD(id: Iri, source: Json): IO[ResourceRejection, (CompactedJsonLd, ExpandedJsonLd)] =
+    for {
+      expandedNoId <- JsonLd.expand(source).leftMap(err => InvalidJsonLdFormat(Some(id), err))
+      id           <- expandedNoId.rootId.asIri match {
+                        case Some(sourceId) if sourceId != id => IO.raiseError(UnexpectedResourceId(id, sourceId))
+                        case _                                => IO.pure(id)
+                      }
+      expanded      = expandedNoId.replaceId(id)
+      compacted    <- expanded.toCompacted(source).leftMap(err => InvalidJsonLdFormat(Some(id), err))
+    } yield (compacted, expanded)
+
+  private def validateSameSchema(
```

Comment:
Shouldn't we have a rejection ?

We can't make a difference if the resource exists or not.

It could be a rejection ending with a 404

Reply:
No, this will be done in the routes. As you can see, the `ResourceNotFound` Rejection has a field where the schema goes